lib/screenplay/alerts/screens_by_alert.ex

```diff
@@ -0,0 +1,48 @@
+defmodule Screenplay.Alerts.ScreensByAlert do
+  @moduledoc """
+  Module used to get ScreensByAlert data from the Screens app.
+  """
+  require Logger
+
+  def get_screens_by_alert(alert_ids) do
+    url = build_url(alert_ids)
+
+    with {:http_request, {:ok, response}} <-
+           {:http_request, HTTPoison.get(url)},
+         {:response_success, %{status_code: 200, body: body}} <-
+           {:response_success, response},
+         {:parse, {:ok, parsed}} <- {:parse, Jason.decode(body)} do
+      {:ok, parsed}
+    else
+      {:http_request, e} ->
+        {:error, httpoison_error} = e
+        log_api_error({:http_fetch_error, e}, message: Exception.message(httpoison_error))
+
+      {:response_success, %{status_code: status_code, body: body}} = response ->
+        _ = log_api_error({:bad_response_code, response}, status_code: status_code, body: body)
+
+        :bad_response_code
```

Comment:
What's the reason for returning this atom in this branch considering that we don't do anything similar in the other error cases? I think the end result would be the same - that is, a no match exception would be raised from the controller, but just curious.

Reply:
There's no good reason for that. I changed all returns to `:error` in the `else` and added a default return in the controller.